src/services/vision/vision.ts

```diff
@@ -0,0 +1,78 @@
+import type { MimeType } from '../../main';
+import type { Resource, StructType } from '../../types';
+import type { Classification, Detection, PointCloudObject } from './types';
+
+/** A service that enables various computer vision algorithms */
+export interface Vision extends Resource {
+  /**
+   * Get a list of detections in the next image given a camera.
+   *
+   * @param cameraName - The name of the camera to use for detection.
+   * @returns - The list of Detections.
+   */
+  getDetectionsFromCamera: (
+    cameraName: string,
+    extra?: StructType
+  ) => Promise<Detection[]>;
+
+  /**
+   * Get a list of detections in the given image.
+   *
+   * @param image - The image from which to get detections.
+   * @param width - The width of the image.
+   * @param height - The height of the image.
+   * @param mimeType - The MimeType of the image.
+   * @returns - The list of Detections.
+   */
+  getDetections: (
+    image: Uint8Array,
+    width: number,
+    height: number,
+    mimeType: MimeType,
```

Comment:
I want to call out that this is different from the way other SDKs do it. Because we're not using a native representation of an image (like `PIL.Image` for python or `img.Image` for golang), we have no way of getting the width/height. So I'm adding them as explicit parameters here.

Cc @bhaney

Reply:
Also, the most important function to get out is `getDetectionsFromCamera`, so if `getDetections` and `getClassifications` aren't 100% because of this issue, it's probably still worth merging this in, and then addressing any of those concerns in a follow-up PR

Reply:
@njooma @maximpertsov not getting the width or the height is a pretty big deal as most vision service functions will require that in some way or another. getDetectionsFromCamera is usually implemented as a wrapper around getDetections and feeds the images it gets from the camera into getDetections. Are there helper functions that can be added to the SDK for reading the headers of JPEG or PNG images, which usually store the dimensions of the image within them?

Reply:
Technically, all images should be able to be deciphered from their headers (height, width, and mime-type) so it probably is not explicitly necessary to have them as arguments into these functions